test/CountdownBarrierTest.php

```diff
@@ -0,0 +1,229 @@
+<?php
+
+namespace Amp\Sync\Test;
+
+use Amp\Loop;
+use Amp\Sync\CountdownBarrier;
+use Error;
+use PHPUnit\Framework\TestCase;
+
+class CountdownBarrierTest extends TestCase
+{
+    public function testSignalUntilResolved(): void
+    {
+        Loop::run(function () {
```

Comment:
You can use `AsyncTestCase` in this package to remove the need to call `Loop::run()` in tests.

Reply:
done